### PR TITLE
Search component

### LINKED ISSUE #23: Search Component and Adopt Modal
image

### PR DESCRIPTION
**Pull request by ayse8888**
Hi,
I built Search Component #23 
Below is my visuals
I used formik Yup for form validation. I added as a package in my branch

1. when page loads
image

2. when user makes a search
image

3. when there is no items matching
image

4. when user clicks adopts button
image

5. when form submitted
image

### CODE REVIEW

**Comment by halitbatur** [comment]
@ayse8888 please update branch and fix conflicts

File: src/components/Search/AdoptionFormModal.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        <Formik
+          initialValues={{
+            firstName: "",
+            lastName: "",
+            email: "",
+            petName: "",
+            petBreed: "",
+            userCity: "",
+          }}
+          validationSchema={Yup.object({
```
Comment (halitbatur) [inline src/components/Search/AdoptionFormModal.js]: can we move this to another variable and then call it here ?

File: src/components/Search/AdoptionFormModal.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            return JSON.stringify(values, null, 2);
+          }}
+        >
+          <Form className="adoptionForm">
+            <div className="errorMessage">
+              <ErrorMessage name="firstName" />
+            </div>
+            <label htmlFor="firstName"></label>
+            <Field name="firstName" type="text" placeholder="Your First Name" />
+
```
Comment (halitbatur) [inline src/components/Search/AdoptionFormModal.js]: extra spaces

File: src/components/Search/AdoptionFormModal.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            <div className="errorMessage">
+              <ErrorMessage name="lastName" />
+            </div>
+            <label htmlFor="lastName"></label>
+            <Field name="lastName" type="text" placeholder="Your Last Name" />
+
```
Comment (halitbatur) [inline src/components/Search/AdoptionFormModal.js]: extra new lines between components

File: src/components/Search/AdoptionFormModal.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            <div className="errorMessage">
+              <ErrorMessage name="email" />
+            </div>
+            <label htmlFor="email"></label>
+            <Field name="email" type="email" placeholder="Your Email Address" />
+
```
Comment (halitbatur) [inline src/components/Search/AdoptionFormModal.js]: extra line between components

File: src/components/Search/AdoptionFormModal.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            <div className="errorMessage">
+              <ErrorMessage name="petName" />
+            </div>
+            <label htmlFor="petName"></label>
+            <Field
+              name="petName"
+              type="text"
+              placeholder="The Name of the Pet You are Adopting"
+            />
+
```
Comment (halitbatur) [inline src/components/Search/AdoptionFormModal.js]: extra new line

File: src/components/Search/AdoptionFormModal.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            <div className="errorMessage">
+              <ErrorMessage name="petBreed" />
+            </div>
+            <label htmlFor="petBreed"></label>
+            <Field
+              name="petBreed"
+              type="text"
+              placeholder="The Breed of the Pet You are Adopting"
+            />
+
```
Comment (halitbatur) [inline src/components/Search/AdoptionFormModal.js]: extra new line

File: src/components/Search/AdoptionFormModal.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+            <div className="errorMessage">
+              <ErrorMessage name="userCity" />
+            </div>
+            <label htmlFor="userCity"></label>
+            <Field name="userCity" type="text" placeholder="Your City" />
+
```
Comment (halitbatur) [inline src/components/Search/AdoptionFormModal.js]: extra new line

File: src/components/Search/Search.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+              name="gender"
+              value={formik.values.gender}
+              {...formik.getFieldProps("gender")}
+            >
+              <option value="Select Gender">Select Gender</option>
+              {searchGuests.searchGender.map((gender) => (
+                <option>{gender}</option>
+              ))}
+            </select>
+
```
Comment (halitbatur) [inline src/components/Search/Search.js]: extra new line

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/Search/AdoptionFormModal.js (hunk 1 above)
Comment (ayse8888) [reply src/components/Search/AdoptionFormModal.js]: done

File: src/components/Search/AdoptionFormModal.js (hunk 2 above)
Comment (ayse8888) [reply src/components/Search/AdoptionFormModal.js]: done

File: src/components/Search/AdoptionFormModal.js (hunk 3 above)
Comment (ayse8888) [reply src/components/Search/AdoptionFormModal.js]: done

File: src/components/Search/AdoptionFormModal.js (hunk 4 above)
Comment (ayse8888) [reply src/components/Search/AdoptionFormModal.js]: done

File: src/components/Search/AdoptionFormModal.js (hunk 5 above)
Comment (ayse8888) [reply src/components/Search/AdoptionFormModal.js]: done

File: src/components/Search/AdoptionFormModal.js (hunk 6 above)
Comment (ayse8888) [reply src/components/Search/AdoptionFormModal.js]: done

File: src/components/Search/AdoptionFormModal.js (hunk 7 above)
Comment (ayse8888) [reply src/components/Search/AdoptionFormModal.js]: done

File: src/components/Search/Search.js (hunk 8 above)
Comment (ayse8888) [reply src/components/Search/Search.js]: done

**Review by Rofy87 — APPROVED** [review]
Thanks for the great work ayse

**Review by mustafaguldag — APPROVED** [review]
It looks solid Ayşe. Thanks for your contribute!

**Review by halitbatur — APPROVED** [review]

**Pull request merged**